backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/TextCheckService.java

```diff
@@ -235,30 +184,55 @@ protected TextCheckCategoryResponse checkCategoryByHTML(
     return new TextCheckCategoryResponse(newHtmlText.toString(), modifiedMatches);
   }
 
-  private List<Match> checkCaseFacts(DocumentationUnit documentationUnit) {
-    if (documentationUnit.longTexts().caseFacts() == null) {
-      return Collections.emptyList();
+  private Match limitReplacements(Match match) {
+    if (match.replacements() != null) {
+      return match.toBuilder()
+          .replacements(match.replacements().subList(0, Math.min(match.replacements().size(), 5)))
+          .build();
     }
-    return checkText(documentationUnit.longTexts().caseFacts(), CategoryType.CASE_FACTS);
+    return match;
   }
 
-  private List<Match> checkDecisionReasons(DocumentationUnit documentationUnit) {
-    if (documentationUnit.longTexts() == null
-        || documentationUnit.longTexts().decisionReasons() == null) {
+  private List<Match> checkText(String text, CategoryType categoryType, UUID documentationUnitId) {
+    if (text == null) {
       return Collections.emptyList();
     }
 
-    return checkText(
-        documentationUnit.longTexts().decisionReasons(), CategoryType.DECISION_REASONS);
+    return checkCategoryByHTML(text, categoryType, documentationUnitId).matches().stream()
+        .map(match -> match.toBuilder().category(categoryType).build())
+        .toList();
   }
 
-  private List<Match> checkText(String text, CategoryType categoryType) {
-    if (text == null) {
-      return Collections.emptyList();
-    }
+  public void removeIgnoredWord(UUID documentationUnitId, String word) {
+    ignoredTextCheckWordRepository.deleteAllByWordAndDocumentationUnitId(word, documentationUnitId);
+  }
 
-    return checkCategoryByHTML(text, categoryType, null, null).matches().stream()
-        .map(match -> match.toBuilder().category(categoryType).build())
+  public IgnoredTextCheckWord addIgnoreWord(UUID documentationUnitId, String word) {
+    return ignoredTextCheckWordRepository.addIgnoredTextCheckWord(word, documentationUnitId);
+  }
+
+  public List<Match> addIgnoredTextChecksIndividually(
+      UUID documentationUnitId, List<Match> matches) {
+    var words = matches.stream().map(Match::word).toList();
+
+    List<IgnoredTextCheckWord> globalAndDocumentationUnitIgnoredWords =
+        ignoredTextCheckWordRepository.findByDocumentationUnitIdOrByGlobalWords(
+            words, documentationUnitId);
+
+    Map<String, List<IgnoredTextCheckWord>> groupedByWord =
+        globalAndDocumentationUnitIgnoredWords.stream()
+            .collect(Collectors.groupingBy(IgnoredTextCheckWord::word));
+
+    if (documentationUnitId == null) {
+      return matches;
+    }
+    return matches.stream()
+        .map(
+            match -> {
+              List<IgnoredTextCheckWord> ignoredWords =
+                  groupedByWord.getOrDefault(match.word(), null);
+              return match.toBuilder().ignoredTextCheckWords(ignoredWords).build();
```

Comment:
I'm thinking: Does it really make sense to send a **list** of ignoredWords for each match? Maybe we can calculate a "winning" ignoreWord (which would probably be the global) and simply remove the other?
Why I stumbled upon this: If we handle a list, we need to handle and render cases where a match is ignored by both jdv/global and decision and I don't see why this would make sense functionally. @Nadav-B - what do you think?

Reply:
Yes, the client needs to know why the word is ignored (currently global/jdv or documentationUnit). If I understand the current implementation correctly, we can give the client multiple reasons. Currently we could express that the word is ignored by both jDV and documentationUnit. 
I'm still not sure if the client can meaningfully use these two reasons (e.g. display two modals?). But maybe I missed something. Maybe let's sync about it